account_payment_term_cutoff_day/models/account_payment_term_line.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    months = fields.Integer(required=True, default=0)
+    end_month = fields.Boolean(
+        compute="_compute_end_month",
+        string="End of month",
+        help="Switch to end of the month after having added months.",
+    )
+    cutoff_day = fields.Integer(
+        default=31,
+        help="Specify the cutoff day of the month for adjusting invoice due dates."
+        "For example, if you set this field to 20, any invoice dated on the 21st or "
+        "later will have its due date moved to the following month."
+        "Setting 0 is treated the same as having no cutoff day, which in practice is "
+        "the same as setting 31.",
+    )
+    has_cutoff_day = fields.Boolean(related="payment_id.has_cutoff_day")
+
+    @api.depends("payment_id.has_cutoff_day")
+    def _compute_end_month(self):
+        for rec in self:
+            rec.end_month = False
+            if rec.payment_id.has_cutoff_day:
+                rec.end_month = True
+
+    def _get_due_date(self, date_ref):
+        self.ensure_one()
+        if date_ref and self.payment_id.has_cutoff_day:
+            date_dt = fields.Date.to_date(date_ref)
+            due_date = date_dt + relativedelta(months=self.months)
+            due_date += relativedelta(day=31)
+            due_date += relativedelta(days=self.nb_days)
+            if self.cutoff_day and date_dt.day > self.cutoff_day:
+                due_date += relativedelta(months=1)
+            return due_date
```

Comment:
We need to reflect the change in https://github.com/OCA/l10n-japan/commit/cb29fa13d3d188b5b8521c13f6d53f57263991b1.